src/app/login/login.component.ts

```diff
@@ -1,21 +1,43 @@
-import { Component } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { Router } from '@angular/router';
 import { CookieService } from 'ngx-cookie-service';
+import { Subscription } from 'rxjs';
+import { User } from 'src/interfaces/User';
+import { UserService } from '../user.service';
 
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
   styleUrls: ['./login.component.css'],
 })
-export class LoginComponent {
+export class LoginComponent implements OnInit, OnDestroy {
   username!: string;
   password!: string;
   errorMessage!: string;
+  users!: User[];
+  subscription = new Subscription();
 
-  constructor(private router: Router, private cookieService: CookieService) {}
+  constructor(
+    private router: Router,
+    private cookieService: CookieService,
+    private userService: UserService
+  ) {}
+
+  ngOnInit(): void {
+    this.userService.getUsers().subscribe((users) => {
+      this.users = users;
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.subscription.unsubscribe();
+  }
 
   logIn() {
-    if (this.username === 'mahomuri' && this.password === 'password123') {
```

Comment:
let's try logging in with a list of dynamic users. not hard coded. use  json to store the list of users that are allowed to login or use your existing user list and add properties for the username and password.

Reply:
Wdym po? It's already dynamic based on the users array from the service. Kindly see `login.component.ts:37`.

Reply:
ok nevermind